server/repo-op.c

```diff
@@ -1896,32 +1932,79 @@ cross_repo_copy (const char *src_repo_id,
         ret = -1;
         goto out;
     }
+    gboolean is_virtual = is_virtual_repo_and_origin (src_repo, dst_repo);
```

Comment:
There should be no need to check whether src and dst is virtual repo and origin. cross_repo_copy() only gets called after this is checked already.